Sources/Sentry/SentryMetricProfiler.mm

```diff
@@ -193,6 +247,7 @@ - (void)recordMemoryFootprint
     }
 
     @synchronized(self) {
+        SENTRY_LOG_DEBUG(@"Recording memory footprint of %llu", footprintBytes);
```

Comment:
yeah this one is super verbose, agreed. it helped me debug some stuff in the tests but i can remove it. 

Maybe it'd be worthwhile introducing another "trace" log level for our purposes, but, i'd be hesitant to do so until  we had macros that didn't even _evaluate_ the log statements beneath the configured log level, to avoid building up huge strings that are eventually discarded. See again, https://github.com/getsentry/sentry-cocoa/issues/4020